x-pack/plugins/infra/public/components/asset_details/asset_details.tsx

```diff
@@ -0,0 +1,228 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useState } from 'react';
+import {
+  EuiFlyout,
+  EuiFlyoutHeader,
+  EuiTitle,
+  EuiFlyoutBody,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiSpacer,
+  EuiTabs,
+  EuiTab,
+  useEuiTheme,
+} from '@elastic/eui';
+import { css } from '@emotion/react';
+import Metadata from './metadata/metadata';
+import { LinkToUptime } from './links/link_to_uptime';
+import { LinkToApmServices } from './links/link_to_apm_services';
+import { Processes } from './processes/processes';
+import type { HostNodeRow } from './types';
+import type { InventoryItemType } from '../../../common/inventory_models/types';
+import type { SetNewHostFlyoutOpen } from '../../pages/metrics/hosts/hooks/use_host_flyout_open_url_state';
+
+export enum FlyoutTabIds {
+  METADATA = 'metadata',
+  PROCESSES = 'processes',
+}
+
+export type TabIds = 'metadata' | 'processes';
+
+export interface Tab {
+  id: FlyoutTabIds;
+  name: any;
+  'data-test-subj': string;
+}
+
+export interface AssetDetailsProps {
+  node: HostNodeRow;
+  nodeType: InventoryItemType;
+  closeFlyout: () => void;
+  renderedTabsSet: React.MutableRefObject<Set<TabIds>>;
+  currentTimeRange: {
+    interval: string;
+    from: number;
+    to: number;
+  };
+  tabs: Tab[];
+  hostFlyoutOpen?: {
+    clickedItemId: string;
+    selectedTabId: TabIds;
+    searchFilter: string;
+    metadataSearch: string;
+  };
+  setHostFlyoutOpen?: SetNewHostFlyoutOpen;
+  onTabClick?: (tab: Tab) => void;
+  links?: Array<'uptime' | 'apmServices'>;
+  showInFlyout?: boolean;
+  showActionsColumn?: boolean;
+}
+
+// Setting host as default as it will be the only supported type for now
+const NODE_TYPE = 'host' as InventoryItemType;
+
+export const AssetDetails = ({
+  node,
+  closeFlyout,
+  onTabClick,
+  renderedTabsSet,
+  currentTimeRange,
+  hostFlyoutOpen,
+  setHostFlyoutOpen,
+  tabs,
+  showInFlyout,
+  links,
+  showActionsColumn,
+  nodeType = NODE_TYPE,
+}: AssetDetailsProps) => {
+  const { euiTheme } = useEuiTheme();
+  const [selectedTabId, setSelectedTabId] = useState('metadata');
```

Comment:
I see, I will think about it - here I wanted to provide 2 options - to have a version that is not relying on the provided state in the `hostFlyoutOpen.selectedTabId` ( that's why it's also optional ) and to have the possibility to keep the state in the component in case it shouldn't rely on url state in some cases and a version similar to host detail flyout where the state is coming from the consumer if we want to persist it outside of the embeddable.